### PR TITLE
docs: add plugin generator info

### PR DESCRIPTION
**Pull request by juanpicado**
- add info about the new plugin generator
- add detailed info about each type of plugin (new pages for each)
- sidebar new navigation
- remove flow notes (old stuff)

🙃 I'm sure tons of typos are around.

### CODE REVIEW

File: docs/plugin-middleware.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+interface IPluginMiddleware<T> extends IPlugin<T> {
+  register_middlewares(app: any, auth: IBasicAuth<T>, storage: IStorageManager<T>): void;
+}
+```
+
+### `register_middlewares`
+
+The method provide full access to the authentification and storage via `auth` and `storage`. `app` is the express application that allows you to add new endpoints.
+
+```
```
Comment (anikethsaha) [inline docs/plugin-middleware.md]:
````suggestion
```typescript
````

File: docs/plugin-storage.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    add(name: string, callback: Callback): void;
+    remove(name: string, callback: Callback): void;
+    get(callback: Callback): void;
+    getSecret(): Promise<string>;
+    setSecret(secret: string): Promise<any>;
+    getPackageStorage(packageInfo: string): IPackageStorage;
+    search(onPackage: onSearchPackage, onEnd: onEndSearchPackage, validateName: onValidatePackage): void;
+  }
+```
+* The `IPackageStorage` is an object is created by each request that handle the I/O actions for the metadata and tarballs.
```
Comment (anikethsaha) [inline docs/plugin-storage.md]:
```suggestion
* The `IPackageStorage` is an object that is created by each request that handles the I/O actions for the metadata and tarballs.
```

File: docs/plugin-storage.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+			updatePackage(
+				pkgFileName: string,
+				updateHandler: StorageUpdateCallback,
+				onWrite: StorageWriteCallback,
+				transformPackage: PackageTransformer,
+				onEnd: CallbackAction
+			): void;
+			savePackage(fileName: string, json: Package, callback: CallbackAction): void;
+  }
+```
```
Comment (anikethsaha) [inline docs/plugin-storage.md]:
```typescript
interface IPackageStorage {
  logger: Logger;
  writeTarball(pkgName: string): IUploadTarball;
  readTarball(pkgName: string): IReadTarball;
  readPackage(fileName: string, callback: ReadPackageCallback): void;
  createPackage(pkgName: string, value: Package, cb: CallbackAction): void;
  deletePackage(fileName: string, callback: CallbackAction): void;
  removePackage(callback: CallbackAction): void;
  updatePackage(
    pkgFileName: string,
    updateHandler: StorageUpdateCallback,
    onWrite: StorageWriteCallback,
    transformPackage: PackageTransformer,
    onEnd: CallbackAction
  ): void;
  savePackage(fileName: string, json: Package, callback: CallbackAction): void;
}
```

I guess this is the formatted code snippet.
I use `prettier's` chrome extension , really helpfull for github related stuff

File: docs/plugin-storage.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+### List Community Storage Plugins
```
Comment (anikethsaha) [inline docs/plugin-storage.md]: I guess the generator link is missing for this page

File: docs/plugin-auth.md [hunk 5]
```diff
@@ -0,0 +1,300 @@
...
+id: plugin-auth
+title: "Authentication Plugin"
+---
+
+## What's an Authentication Plugin?
+
+Is a sort plugin that allows to handle who access or publish to a specific package. By default the `htpasswd` is built-in, but can
+easily be replaced by your own.
+
+<div id="codefund">''</div>
```
[the remark this answers is not in the document]
Comment (juanpicado) [reply docs/plugin-auth.md]: Yes is for the ads :) unique thing allows us to make money and pay the servers

[the remark this answers is not in the document]
Comment (anikethsaha) [reply docs/plugin-auth.md]: > the ads :) unique thing allows us to make money and pay the

yeah but that `''` quotes inside the dive is required?

File: docs/plugin-auth.md [hunk 6]
```diff
@@ -0,0 +1,300 @@
...
+import { getNotFound } from '@verdaccio/commons-api';
+
+ const err = getForbidden('not allowed to access package');
+
+callback(err);
+```
+
+## Generate an authentication plugin
+
+For detailed info check our [plugin generator page](plugin-generator). Run the `yo` command in your terminal and follow the steps.
```
[the remark this answers is not in the document]
Comment (juanpicado) [reply docs/plugin-auth.md]: The syntax is used by docusaurus, I doubt is GitHub friendly you must run the website locally

[the remark this answers is not in the document]
Comment (anikethsaha) [reply docs/plugin-auth.md]: that only I doubted, actually. cool 👍

File: docs/plugin-auth.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (juanpicado) [reply docs/plugin-auth.md]: yes

File: docs/plugin-storage.md (hunk 3 above)
Comment (juanpicado) [reply docs/plugin-storage.md]: I must do it locally, I tried hopefully now looks better.

**Comment by juanpicado** [comment]
Thank guys !

**Pull request merged**